plugins/grid-bookmark/src/GridBookmarkWidget/model.ts

```diff
@@ -159,64 +146,59 @@ export default function f(_pluginManager: PluginManager) {
         bookmark: IExtendedLabeledRegionModel,
         label: string,
       ) {
-        const target = self.bookmarkedRegions.find(
-          (element: ILabeledRegionModel) => {
-            return element === bookmark.correspondingObj
-          },
-        )
-        target?.setLabel(label)
+        bookmark.correspondingObj.setLabel(label)
       },
       setSelectedBookmarks(bookmarks: IExtendedLabeledRegionModel[]) {
         self.selectedBookmarks = bookmarks
       },
       setBookmarkedRegions(
         bookmarkedRegions: IMSTArray<typeof LabeledRegionModel>,
       ) {
-        self.bookmarkedRegions = bookmarkedRegions
+        self.bookmarks = cast(bookmarkedRegions)
       },
     }))
     .actions(self => ({
-      updateLocalStorage() {
-        localStorageSetItem(
-          self.localStorageKey,
-          JSON.stringify(self.bookmarkedRegions),
-        )
-      },
       clearAllBookmarks() {
         self.bookmarkedRegions.forEach(bookmark => {
           if (self.validAssemblies.has(bookmark.assemblyName)) {
-            self.bookmarkedRegions.remove(bookmark)
+            self.bookmarks?.remove(bookmark)
           }
         })
-        this.updateLocalStorage()
-        self.updateSelectedAssembliesAfterClear()
       },
       clearSelectedBookmarks() {
         self.selectedBookmarks.forEach(selectedBookmark => {
-          self.bookmarkedRegions.remove(selectedBookmark.correspondingObj)
+          self.bookmarks?.remove(selectedBookmark.correspondingObj)
         })
         self.selectedBookmarks = []
-        this.updateLocalStorage()
-        self.updateSelectedAssembliesAfterClear()
       },
+    }))
+    .actions(self => ({
       afterAttach() {
+        const key = localStorageKeyF()
         addDisposer(
           self,
           autorun(() => {
-            if (self.bookmarkedRegions.length > 0) {
-              localStorageSetItem(
-                self.localStorageKey,
-                JSON.stringify(self.bookmarkedRegions),
-              )
-            } else {
+            if (self.bookmarks === undefined) {
+              // load bookmarks from localstorage, note that this is the primary route for
+              // loading bookmarks, as they are removed from session snapshots
```

Comment:
crucially, the bookmarks are ONLY loaded from localstorage here instead of session state, by this autorun